trace/noop.go

```diff
@@ -19,16 +19,20 @@ import (
 
 	"go.opentelemetry.io/otel/attribute"
 	"go.opentelemetry.io/otel/codes"
+	"go.opentelemetry.io/otel/trace/embedded"
 )
 
 // NewNoopTracerProvider returns an implementation of TracerProvider that
 // performs no operations. The Tracer and Spans created from the returned
 // TracerProvider also perform no operations.
+//
+// Deprecated: Use [go.opentelemetry.io/otel/trace/noop.NewTracerProvider]
```

Comment:
I'm pretty sure this was discussed at the last SIG meeting, but i've forgotten.  Why is the no-op being moved?

Reply:
We need to export concrete No-Op types so users can embedded them in their implementation if they choose they want no-op behavior as the default. Currently, the API does not export any concrete No-Op types. So these types need to be added in `otel/trace` or they can be added in a new `otel/trace/noop`

If they are added in `otel/trace`:

- They will all need some way to signify they are no-op types, meaning they will all need some prefix of `Noop`
  - This prefix is the using names to scope instead of a package
- This layout will not match the metric API structure that has its No-Op types in a `metric/noop` package
- `NewNoopTracerProvider` will not need to be deprecated

If they are added in `otel/trace/noop`:

- They will all be scoped via a package and can use type names without prefixes
- The package structure will match the metric API
- It will require the `NewNoopTracerProvider` function in `otel/trace` be deprecated and replaced with one in the `noop` package
- Unexported types in `otel/trace` are not accessible
  - The `Start` method cannot type assert a span is a `nonRecordingSpan`, but this seems like a bug fix as it will instead check for a valid span context and if the span is recording more generally.

Given the shorter more concise naming and symmetry with the metric API, creating the `noop` package option was chosen.